pkg/nodeconfig/payload/payload.go

```diff
@@ -110,6 +110,36 @@ param(
     }
   }
 
+# This retries getting the HNS-network until it succeeds
+function Retry-GetHnsNetwork {
+     param (
+         [int]$retryCount = 20,
+         [int]$retryDelay = 10
+     )
+ 
+     $attempt = 0
+ 
+     while ($attempt -lt $retryCount) {
+         try {
+             $hns_network = Get-HnsNetwork | Where-Object { $_.Name -eq 'HNS_NETWORK' }
+             
+             # Check if hns_network is null
+             if ($null -eq $hns_network) {
+                 Write-Host "Attempt $($attempt + 1) returned null. Retrying..."
+             } else {
+                 Write-Host "Found HNS_NETWORK on attempt $($attempt + 1)"
+                 return $hns_network 
+             }
+         } catch {
+             Write-Host "Attempt $($attempt + 1) failed: $_"
+         }
+         Start-Sleep -Seconds $retryDelay
+         $attempt++
+     }
+     Write-Host "Max retry attempts reached. Continuing."
```

Comment:
I feel dropping `Continuing` would be apt here
In case of a timeout, in the logs you would see `Continuing` and then the fatal error, which is not really continuing :laughing:

Reply:
FAIR, LOL